src/test/ts/core/NpmTagsTest.ts

```diff
@@ -233,5 +248,22 @@ describe('NpmTags', () => {
         });
       });
     });
+
+    describe('arbitrary branches', () => {
+      const specialChar = () => fc.char().filter((char) => !/[\w.]/.test(char));
+      const alphanumeric = () => fc.char().filter((char) => /\w/.test(char));
+
+      it('arbitrary branch names should collapse special characters', () =>
```

Comment:
```suggestion
      it('should collapse special characters', () =>
```
Should we consider changing the describe to "arbitrary branch names" so that the sentence flow stays equivalent?